src/dist/replication/meta_server/meta_service.cpp

```diff
@@ -603,30 +603,13 @@ void meta_service::on_query_cluster_info(configuration_cluster_info_rpc rpc)
 void meta_service::on_query_configuration_by_index(configuration_query_by_index_rpc rpc)
 {
     configuration_query_by_index_response &response = rpc.response();
-
-    // here we do not use RPC_CHECK_STATUS macro, but specially handle it
-    // to response forward address.
-    dinfo("rpc %s called", __FUNCTION__);
     rpc_address forward_address;
-    int result = check_leader(rpc.dsn_request(), &forward_address);
-    if (result == 0) {
-        rpc.disable_auto_reply();
-        return;
-    }
-    if (result == -1 || !_started) {
-        if (result == -1) {
-            response.err = ERR_FORWARD_TO_OTHERS;
-            if (!forward_address.is_invalid()) {
-                partition_configuration config;
-                config.primary = forward_address;
-                response.partitions.push_back(std::move(config));
-            }
-        } else if (_recovering) {
-            response.err = ERR_UNDER_RECOVERY;
-        } else {
-            response.err = ERR_SERVICE_NOT_ACTIVE;
+    if (!check_status(rpc, &forward_address)) {
+        if (!forward_address.is_invalid()) {
```

Comment:
What if `check_leader` returns 0?  `check_status` will then return false, with no change to `forward_address`. 
Since that `forward_address` is invalid by default, it will come into this logic block.

```cpp
void meta_service::on_query_configuration_by_index(configuration_query_by_index_rpc rpc)
{
    configuration_query_by_index_response &response = rpc.response();
    rpc_address forward_address;
    if (!check_status(rpc, &forward_address)) { // returns false when check_leader returns 0
        if (!forward_address.is_invalid()) { // invalid by default
            partition_configuration config; // formerly it returns directly, without coming into this logic
            config.primary = forward_address;
            response.partitions.push_back(std::move(config));
        }
        return;
    }
```

In this situation, we formerly return directly (because the RPC is already forwarded). But now you make it responds to the client with the leader address.

```cpp
void meta_service::on_query_configuration_by_index(configuration_query_by_index_rpc rpc)
{
    configuration_query_by_index_response &response = rpc.response();
    rpc_address forward_address;
    int result = check_leader(rpc.dsn_request(), &forward_address);
    if (result == 0) { //  check_leader returns 0.
        rpc.disable_auto_reply();
        return; // it returns directly
    }
```

Is this change reasonable, or an unconsidered case?